common/rpc/encryption/localStoreTlsProvider.go

```diff
@@ -225,31 +229,33 @@ func (s *localStoreTlsProvider) getOrCreateConfig(
 func newServerTLSConfig(
 	certProvider CertProvider,
 	perHostCertProviderMap PerHostCertProviderMap,
+	config *config.GroupTLS,
 ) (*tls.Config, error) {
 
-	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider)
+	clientAuthRequired := config.Server.RequireClientAuth
+	tlsConfig, err := getServerTLSConfigFromCertProvider(certProvider, clientAuthRequired)
 	if err != nil {
 		return nil, err
 	}
 
 	tlsConfig.GetConfigForClient = func(c *tls.ClientHelloInfo) (*tls.Config, error) {
 		if perHostCertProviderMap != nil {
-			perHostCertProvider, err := perHostCertProviderMap.GetCertProvider(c.ServerName)
+			perHostCertProvider, clientAuth, err := perHostCertProviderMap.GetCertProvider(c.ServerName)
 			if err != nil {
 				return nil, err
 			}
 
 			if perHostCertProvider == nil {
```

Comment:
unrelated to your change, but is this easier to read if we just do:

if perHostCertProvider != nil {
    return getServerTLSConfigFromCertProvider(perHostCertProvider, clientAuth)
}

return getServerFromTLSConfigProvider(certProvider, clientAuthRequired)

Helps get rid of a return statement